pkg/controller/core/workload_controller.go

```diff
@@ -169,7 +169,11 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 
 	if len(wl.OwnerReferences) == 0 && !wl.DeletionTimestamp.IsZero() {
 		// manual deletion triggered by the user
-		return ctrl.Result{}, workload.RemoveFinalizer(ctx, r.client, &wl)
+		err := workload.RemoveFinalizer(ctx, r.client, &wl)
+		if client.IgnoreNotFound(err) != nil {
+			return ctrl.Result{}, err
+		}
+		return ctrl.Result{}, nil
```

Comment:
```suggestion
			err := workload.ApplyAdmissionStatus(ctx, r.client, &wl, true, r.clock)
			return ctrl.Result{}, client.IgnoreNotFound(err)
```

And also other places.

Reply:
Done.